Flash card landing page

tonyly98 opened the pull request:
Hey! I'm pretty new to TypeScript and Remix, please let me know if I need to fix/change anything!
This PR is for the [flashcard landing page issue](https://github.com/pbruvoll/tekkendocs/issues/161) also includes some typos fixes

### Changes ###
- Created landing pages for flash cards
- Added heading on main page to flash cards landing page
- Fixed typos in challenges page+ flash cards pages

![flashcards-landing-page](https://github.com/user-attachments/assets/e427a606-79e9-4522-8632-674f93c1e21d)

pbruvoll commented:
Thanks for the pull request. It looks great and I love that you exported correct meta data from the new route. Also great you fixed some typos!
Could you make 3 small adjustments?

1. Set cache control header from the loader such that the page is cached correctly in the CDN. You can just do it like its done for the t8 route :
```
export const loader = async (): Promise<TypedResponse<GamePageData>> => {
  return json<GamePageData>(
    { characterInfoList: getTekken8Characters() },
    {
      headers: getCacheControlHeaders({ seconds: 60 * 5 }),
    },
  )
}
```

2. I haven't set up formatting on commit, and I see that the format-check github action is failing. You can run formatting with "npm run format"

3. Can you aadd just a bit more space between the text and the CharacterGrid ?

tonyly98 commented:
Thanks for the feedback! Looking forward to contributing more 😄
Updated the PR:

1. Set cache control header from the loader 
2. Ran `npm run format` and file is now passing `npm run formatcheck` 
3. Added space between the text and CharacterGrid

pbruvoll commented:
Excellent. I was actually thinking about some more visual space in the webpage between text and character grid. Sorry I wasn't more specific. But we can change that later, let's get this PR merged and released.
I am also looking forward to more contributions :)
